src/Agent.Worker/Handlers/Handler.cs

```diff
@@ -20,10 +20,19 @@ public abstract class Handler : AgentService
     {
         protected readonly Dictionary<string, string> Environment = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Why did I make this a field? I believe that's a minor sin when it comes to base classes.